packages/node_modules/@ciscospark/internal-plugin-wdm/test/unit/spec/device.js

```diff
@@ -85,6 +101,133 @@ describe(`plugin-wdm`, () => {
           assert.becomes(device.getServiceUrl(`valid`), `http://example.com/valid-service`)
         ]);
       });
+
+      describe(`when web-ha-messasing is enabled`, () => {
+        it(`returns the service url with a different host if available`, () => {
```

Comment:
What happens when we've run out of hosts? What happens if there's only one host?

Reply:
When we've ran out of hosts, it's going to call `this.refresh()`, this logic and the logic to cycle host is in `markUrlFailedAndGetNew`

Here, for `getServiceUrl`, it'll just use the current host

Reply:
gotcha. thanks!